src/msg_handler.rs

```diff
@@ -10,17 +10,24 @@ use crate::user::committee::Committee;
 use hex::ToHex;
 use std::fmt::Debug;
 
+pub struct HandleMsgOutput {
+    pub result: Message,
+
+    // if true, it instructs the event loop to bubble up the message and terminate.
+    pub is_final_msg: bool,
+}
+
```

Comment:
Why not use an Enum?

A sort of 
```rust
pub enum HandleMsgOutput {
    Result(Message),
    FinalResult(Message)
}
```

Looking at your code seems to fit enough and it's more ideomatic

Reply:
That would be indeed more idiomatic. I'll address it next PR.

Reply:
Actually, I liked that a lot and  I addressed it with this PR. Thanks!